src/AstGen.zig

```diff
@@ -5347,6 +5349,7 @@ fn forExpr(
             .name = index_name,
             .ptr = index_ptr,
             .token_src = index_token,
+            .is_comptime = parent_gz.force_comptime,
```

Comment:
Is this correct?

Reply:
This should use `is_inline` since the index is also comptime known for inline loops. Also sorry for not answering sooner.

Reply:
Btw @andrewrk, @SpexGuy What should the correct behavior be for
```zig
pub fn main() void {
    comptime var i = 0;
    inline while (i < 5) : (i+=1) {
        const S = struct {
            fn print() void {
                std.debug.print("loop {d}\n", .{i});
            }
        };
        S.print();
    }
}
```

Stage1 prints
```
loop 5
loop 5
loop 5
loop 5
loop 5
```
but I think most people would assume it to print
```
loop 0
loop 1
loop 2
loop 3
loop 4
```